### ISSUE TITLE
generator: Cannot find oid 'laLoadInt' to walk

### ISSUE THREAD

**Issue by jpds** [prometheus/snmp_exporter#340]
Following the guide at:

- https://github.com/prometheus/snmp_exporter/tree/master/generator

Did:

```
$ docker run --rm -it golang:1.11 bash
$ apt-get update
$ apt-get install build-essential libsnmp-dev unzip
$ go get github.com/prometheus/snmp_exporter/generator
$ cd ${GOPATH-$HOME/go}/src/github.com/prometheus/snmp_exporter/generator
$ go build
$ make mibs
$ export MIBDIRS=$(pwd)/mibs
$ ./generator generate
INFO[0000] Loading MIBs from /go/src/github.com/prometheus/snmp_exporter/generator/mibs  source="net_snmp.go:136"
WARN[0001] NetSNMP reported 513 parse errors             source="main.go:100"
WARN[0001] Can't find augmenting oid tokenRingMLStatsEntry for tokenRingMLStats2Entry  source="tree.go:81"
WARN[0001] Can't find augmenting oid tokenRingPStatsEntry for tokenRingPStats2Entry  source="tree.go:81"
WARN[0001] Can't find augmenting oid rptrPortEntry for ciscoRptrPortEntry  source="tree.go:81"
INFO[0001] Generating config for module if_mib_ifalias   source="main.go:49"
INFO[0001] Generated 42 metrics for module if_mib_ifalias  source="main.go:60"
INFO[0001] Generating config for module if_mib_ifname    source="main.go:49"
INFO[0001] Generated 42 metrics for module if_mib_ifname  source="main.go:60"
INFO[0001] Generating config for module apcups           source="main.go:49"
INFO[0001] Generated 193 metrics for module apcups       source="main.go:60"
INFO[0001] Generating config for module paloalto_fw      source="main.go:49"
INFO[0002] Generated 152 metrics for module paloalto_fw  source="main.go:60"
INFO[0002] Generating config for module arista_sw        source="main.go:49"
INFO[0002] Generated 90 metrics for module arista_sw     source="main.go:60"
INFO[0002] Generating config for module synology         source="main.go:49"
FATA[0002] Cannot find oid 'laLoadInt' to walk           source="tree.go:288"

**Comment by jpds** [comment]
`generator` seems to randomize the ordering of the MIBs? Also getting:

```
./generator generate 
INFO[0000] Loading MIBs from /go/src/github.com/prometheus/snmp_exporter/generator/mibs  source="net_snmp.go:136"
WARN[0001] NetSNMP reported 512 parse errors             source="main.go:100"
WARN[0001] Can't find augmenting oid tokenRingMLStatsEntry for tokenRingMLStats2Entry  source="tree.go:81"
WARN[0001] Can't find augmenting oid tokenRingPStatsEntry for tokenRingPStats2Entry  source="tree.go:81"
WARN[0001] Can't find augmenting oid rptrPortEntry for ciscoRptrPortEntry  source="tree.go:81"
INFO[0001] Generating config for module ubiquiti_unifi   source="main.go:49"
INFO[0001] Generated 96 metrics for module ubiquiti_unifi  source="main.go:60"
INFO[0001] Generating config for module if_mib           source="main.go:49"
INFO[0001] Generated 42 metrics for module if_mib        source="main.go:60"
INFO[0001] Generating config for module apcups           source="main.go:49"
INFO[0001] Generated 193 metrics for module apcups       source="main.go:60"
INFO[0001] Generating config for module paloalto_fw      source="main.go:49"
INFO[0001] Generated 152 metrics for module paloalto_fw  source="main.go:60"
INFO[0001] Generating config for module ddwrt            source="main.go:49"
FATA[0001] Cannot find oid '1.3.6.1.4.1.2021.10.1.1' to walk  source="tree.go:288"
```

**Comment by jpds** [comment]
The `generator` works with these changes to `generator.yml`:

```
diff --git a/generator/generator.yml b/generator/generator.yml
index db3bfbd..6629989 100644
--- a/generator/generator.yml
+++ b/generator/generator.yml
@@ -142,12 +142,6 @@ modules:
       - interfaces
       - sysUpTime
       - ifXTable
-      - laNames
-      - laLoadInt
-      - ssCpuUser
-      - ssCpuSystem
-      - ssCpuIdle
-      - memory
       - hrStorage
       - 1.3.6.1.4.1.6574.1       # synoSystem
       - 1.3.6.1.4.1.6574.2       # synoDisk
@@ -189,11 +183,6 @@ modules:
       - interfaces
       - ifXTable
       - 1.3.6.1.2.1.25.2 # hrStorage
-      - 1.3.6.1.4.1.2021.4 # memory
-      - 1.3.6.1.4.1.2021.10.1.1 # laIndex
-      - 1.3.6.1.4.1.2021.10.1.2 # laNames
-      - 1.3.6.1.4.1.2021.10.1.5 # laLoadInt
-      - 1.3.6.1.4.1.2021.11 # systemStats
     lookups:
     - old_index: ifIndex
       new_index: ifDescr
```

**Comment by ykfq** [comment]
Same error:
```
[root@lvs01 generator]#  export MIBDIRS=$HOME/.snmp/mibs:mibs:/usr/share/snmp/mibs
[root@lvs01 generator]# ./generator generate
INFO[0000] Loading MIBs from /root/.snmp/mibs:mibs:/usr/share/snmp/mibs  source="net_snmp.go:136"
WARN[0009] NetSNMP reported 527 parse errors             source="main.go:102"
WARN[0011] Can't find augmenting oid tokenRingMLStatsEntry for tokenRingMLStats2Entry  source="tree.go:81"
WARN[0011] Can't find augmenting oid tokenRingPStatsEntry for tokenRingPStats2Entry  source="tree.go:81"
WARN[0011] Can't find augmenting oid rptrPortEntry for ciscoRptrPortEntry  source="tree.go:81"
...
```
but the snmp.yml generated anyway.

**Comment by sgissi** [comment]
These are part of UCD-SNMP-MIB which should have come from your OS. Can you give more details on the your setup (OS, installed SNMP packages, etc)?

**Comment by AlbertasG** [comment]
Try following code to see which MIBs are imported and parsed.
`snmpwalk -D parse-mibs -v <version> -c <key> <ip> system`

As well as running translation & reverse translation
`snmptranslate -On <MIB>::<OID>` 

`snmptraslate -Of <1.3.6.1.4 OID>`

**Comment by daenney** [comment]
This has been fixed with 0.15.0, the Makefile now pulls UCD-SNMP-MIB.
